compiler/z/codegen/BinaryEvaluator.cpp

```diff
@@ -1048,20 +1048,20 @@ lDivRemGenericEvaluator(TR::Node * node, TR::CodeGenerator * cg, bool isDivision
       TR::Register * tempReg4 = cg->allocateRegister();
       TR::Register * tempReg5 = cg->allocateRegister();
       TR::Register * tempReg6 = cg->allocateRegister();
-      dependencies->addPostCondition(tempReg1, TR::RealRegister::GPR0);
-      dependencies->addPostCondition(tempReg2, TR::RealRegister::GPR1);
-      dependencies->addPostCondition(tempReg3, TR::RealRegister::GPR9);
-      dependencies->addPostCondition(tempReg4, TR::RealRegister::GPR8);
+      dependencies->addPostCondition(tempReg1, (TR::RealRegister::RegDep)TR::RealRegister::GPR0);
+      dependencies->addPostCondition(tempReg2, (TR::RealRegister::RegDep)TR::RealRegister::GPR1);
+      dependencies->addPostCondition(tempReg3, (TR::RealRegister::RegDep)TR::RealRegister::GPR9);
+      dependencies->addPostCondition(tempReg4, (TR::RealRegister::RegDep)TR::RealRegister::GPR8);
 
-      dependencies->addPostCondition(evenRegister, TR::RealRegister::GPR2);
+      dependencies->addPostCondition(evenRegister, (TR::RealRegister::RegDep)TR::RealRegister::GPR2);
 
-      dependencies->addPostCondition(oddRegister, TR::RealRegister::GPR3);
-      dependencies->addPostCondition(tempReg5, TR::RealRegister::GPR10);
-      dependencies->addPostCondition(tempReg6, TR::RealRegister::GPR11);
+      dependencies->addPostCondition(oddRegister, (TR::RealRegister::RegDep)TR::RealRegister::GPR3);
+      dependencies->addPostCondition(tempReg5, (TR::RealRegister::RegDep)TR::RealRegister::GPR10);
+      dependencies->addPostCondition(tempReg6, (TR::RealRegister::RegDep)TR::RealRegister::GPR11);
 
       // Setup regs for call out
-      dependencies->addPostCondition(EPReg, cg->getEntryPointRegister());
-      dependencies->addPostCondition(RAReg, cg->getReturnAddressRegister());
+      dependencies->addPostCondition(EPReg, (TR::RealRegister::RegDep)cg->getEntryPointRegister());
+      dependencies->addPostCondition(RAReg, (TR::RealRegister::RegDep)cg->getReturnAddressRegister());
```

Comment:
I don't think delegating these casts to the callee is a good idea. Do you agree that a much less invasive change would be to define two overloads to `addPostCondition` (and similar APIs) and hide these casts under the hood. Once we get scoped enum support all of this can be cleaned up.

Reply:
Yup, that sounds a lot cleaner. I've backed out the casts and introduced the new APIs in 
42a4102 to 6da2dce